crates/test-programs/src/bin/preview1_stdio.rs

```diff
@@ -5,7 +5,8 @@ use test_programs::preview1::{STDERR_FD, STDIN_FD, STDOUT_FD};
 unsafe fn test_stdio() {
     for fd in &[STDIN_FD, STDOUT_FD, STDERR_FD] {
         wasip1::fd_fdstat_get(*fd).expect("fd_fdstat_get on stdio");
-        wasip1::fd_renumber(*fd, *fd + 100).expect("renumbering stdio");
+        assert_eq!(wasip1::fd_renumber(*fd, *fd + 100), Err(wasip1::ERRNO_BADF));
```

Comment:
I don't know much about this implementation, but shouldn't this succeed as it did previously? Doesn't the logic in [`descriptor.rs`](https://github.com/bytecodealliance/wasmtime/pull/11560/files#diff-dbbe655bc4c663438bc5b239e2fc7f27ae063d649a74078701d1446f0f8aa654R366) below expand the table so this _should_ work?

Reply:
Ok, so perhaps we no longer should expand the table? Probably someone more familiar with this should have an opinion here... Feel free to merge if I'm nitpicking here.

Reply:
This is something we can eventually clean up in the future but the extend-the-table code shouldn't ever get hit any more because both to/from are guaranteed to be allocated. In that sense it's just dynamically dead code right now but otherwise harmless